### DISCUSSION TITLE
problems update

### DISCUSSION THREAD

**Discussion by YuriFontella** [dymmond/databasez#39] [General]
hello,

I was using version 0.7.2 performing raw queries with asyncmy
something like:
```py
query_downloads = "SELECT * FROM downloads ORDER BY date desc LIMIT 1"
rows = await db.fetch_all(query_downloads)
return [dict(i) for i in rows]
```

the query returned within a Record class
```
[Record(22, 20000, 0, '["ddd", "tel"]', 'banco5', None, 'file-2024-08-01_17-01-50.csv', 20000, datetime.datetime(2024, 8, 1, 17, 1, 50))]
```
with for I already mapped the columns with the result
```
print([dict(i) for i in rows])
[{'id': 22, 'limit': 20000, 'offset': 0, 'columns': '["ddd", "tel"]', 'table': 'banco5', 'parameters': None, 'file': 'file-2024-08-01_17-01-50.csv', 'rows': 20000, 'date': datetime.datetime(2024, 8, 1, 17, 1, 50)}]
```

when updating to version 0.9.2 the results started coming inside tuples, which ended up breaking all my queries with the error:
```cannot convert dictionary update sequence element #0 to a sequence```

UPDATE

- Removing nest-asyncio (1.6.0)
- Installing orjson (3.10.7)
- Installing sqlalchemy-utils (0.41.2)
- Updating databasez (0.7.2 -> 0.9.2)

return of queries with version 0.9.2:
```
[(22, 20000, 0, '["ddd", "tel"]', 'banco5', None, 'file-2024-08-01_17-01-50.csv', 20000, datetime.datetime(2024, 8, 1, 17, 1, 50))]
```

I downgraded to version 0.7.2
Was this a problem with the package?

**Comment by tarsil** [comment]
Hey @YuriFontella so from 0.7.2 to this one there was an internal rewritten in order to modernize but we will have a look @devkral . Let's address this

**Reply by devkral** [reply]
sorry for the late answer. I reworked the internals again and made them threadsafe. I am not sure but probably the problem will be gone in 0.10.3
